MultiThreading.Task4.Threads.Join/Program.cs

```diff
@@ -10,11 +10,15 @@
  */
 
 using System;
+using System.Threading;
 
 namespace MultiThreading.Task4.Threads.Join;
 
 class Program
 {
+    private static readonly Semaphore _mainSemaphore = new(initialCount: 0, maximumCount: 1);
```

Comment:
You can remove this one easily, plays no role.

Reply:
This is a synchronization semaphore of main function. If I remove all _mainSemaphore calls, I'll see "Done" in command line before all recursive calls will finish.